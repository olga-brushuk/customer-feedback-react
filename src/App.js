import React from 'react';
import './App.css';

class CustomerFeedback extends React.Component {
  constructor(props) {
    super(props);
    this.state = { // set default states
      formControls: {
          email: {
            value: ''
          },
          name: {
            value: ''
          },
          comment: {
            value: ''
          }
      }
    };

    //this.handleNameChange = this.handleNameChange.bind(this);  
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  changeHandler = event => {
      
      const name = event.target.name;
      const value = event.target.value;
    
      this.setState({
        formControls: {
            ...this.state.formControls,
            [name]: {
            ...this.state.formControls[name],
            value
          }
        }
      });
  }

  handleSubmit(event) {
    alert('Email: ' + this.state.formControls.email.value);
    alert('Name: ' + this.state.formControls.name.value);
    alert('Comment: ' + this.state.formControls.comment.value);
    //alert('Comment: ' + this.state.comment);
    event.preventDefault();
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>      
        <label for="name" className="hidden">Enter your email</label>
        <input type="text" name="name" value={this.state.formControls.name.value} onChange={this.changeHandler} placeholder="Enter your name" />
        
        <label for="email" className="hidden">Enter your name</label>
        <input type="email" name="email" value={this.state.formControls.email.value} onChange={this.changeHandler} placeholder="Enter your email" />
        
        <label for="comment" className="hidden">Enter your comment</label>
        <textarea name="comment" value={this.state.formControls.comment.value} onChange={this.changeHandler} placeholder="Please leave your comment" />
        
        <input type="submit" value="Submit" />
      </form>
    );
  }
}

function App() {
  return (
    <CustomerFeedback />
  );
}

export default App;
