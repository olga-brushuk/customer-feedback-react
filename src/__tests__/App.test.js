import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer'

describe('Customer Feedback snapshot',()=>{
	test('testing app initial state', () => {
		let tree = create(<App />)
		expect(tree.toJSON()).toMatchSnapshot();
	})
})