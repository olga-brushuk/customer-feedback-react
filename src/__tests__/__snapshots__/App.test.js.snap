// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Feedback snapshot testing app initial state 1`] = `
<div
  className="slab"
>
  <div
    className="container"
  >
    <h1
      className="below-16"
    >
      Customerr feedback
    </h1>
    <div
      className="row"
    >
      <div
        className="col-md-7"
      >
        <form
          onSubmit={[Function]}
        >
          <label
            className="hidden"
            htmlFor="userName"
          >
            Enter your email
          </label>
          <input
            name="userName"
            onChange={[Function]}
            placeholder="Enter your name"
            required={true}
            type="text"
            value=""
          />
          <label
            className="hidden"
            htmlFor="email"
          >
            Enter your name
          </label>
          <input
            name="email"
            onChange={[Function]}
            placeholder="Enter your email"
            required={true}
            type="email"
            value=""
          />
          <div
            className="below-16 top-24"
          >
            <h2
              className="mega"
            >
              Please include your rating:
            </h2>
            <div
              className="dv-star-rating"
              style={
                Object {
                  "display": "inline-block",
                  "position": "relative",
                }
              }
            >
              <input
                checked={false}
                className="dv-star-rating-input"
                id="rate1_5"
                name="rate1"
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                    "marginLeft": -9999,
                    "position": "absolute",
                  }
                }
                type="radio"
                value={5}
              />
              <label
                className="dv-star-rating-star dv-star-rating-empty-star"
                htmlFor="rate1_5"
                onClick={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "color": "#333",
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              >
                <i
                  style={
                    Object {
                      "fontStyle": "normal",
                    }
                  }
                >
                  ★
                </i>
              </label>
              <input
                checked={false}
                className="dv-star-rating-input"
                id="rate1_4"
                name="rate1"
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                    "marginLeft": -9999,
                    "position": "absolute",
                  }
                }
                type="radio"
                value={4}
              />
              <label
                className="dv-star-rating-star dv-star-rating-empty-star"
                htmlFor="rate1_4"
                onClick={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "color": "#333",
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              >
                <i
                  style={
                    Object {
                      "fontStyle": "normal",
                    }
                  }
                >
                  ★
                </i>
              </label>
              <input
                checked={false}
                className="dv-star-rating-input"
                id="rate1_3"
                name="rate1"
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                    "marginLeft": -9999,
                    "position": "absolute",
                  }
                }
                type="radio"
                value={3}
              />
              <label
                className="dv-star-rating-star dv-star-rating-empty-star"
                htmlFor="rate1_3"
                onClick={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "color": "#333",
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              >
                <i
                  style={
                    Object {
                      "fontStyle": "normal",
                    }
                  }
                >
                  ★
                </i>
              </label>
              <input
                checked={false}
                className="dv-star-rating-input"
                id="rate1_2"
                name="rate1"
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                    "marginLeft": -9999,
                    "position": "absolute",
                  }
                }
                type="radio"
                value={2}
              />
              <label
                className="dv-star-rating-star dv-star-rating-empty-star"
                htmlFor="rate1_2"
                onClick={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "color": "#333",
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              >
                <i
                  style={
                    Object {
                      "fontStyle": "normal",
                    }
                  }
                >
                  ★
                </i>
              </label>
              <input
                checked={true}
                className="dv-star-rating-input"
                id="rate1_1"
                name="rate1"
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                    "marginLeft": -9999,
                    "position": "absolute",
                  }
                }
                type="radio"
                value={1}
              />
              <label
                className="dv-star-rating-star dv-star-rating-full-star"
                htmlFor="rate1_1"
                onClick={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "color": "#ffb400",
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              >
                <i
                  style={
                    Object {
                      "fontStyle": "normal",
                    }
                  }
                >
                  ★
                </i>
              </label>
            </div>
          </div>
          <label
            className="hidden"
            htmlFor="comment"
          >
            Enter your comment
          </label>
          <textarea
            name="comment"
            onChange={[Function]}
            placeholder="Please leave your comment"
            required={true}
            value=""
          />
          <input
            type="submit"
            value="Submit"
          />
        </form>
      </div>
      <div
        className="col-md-5"
      >
        
      </div>
    </div>
    
  </div>
</div>
`;
